contracts/connection-router/src/state.rs

```diff
@@ -101,6 +102,18 @@ pub struct Message {
     pub payload_hash: [u8; 32],
 }
 
+impl Message {
+    pub fn hash_id(&self) -> [u8; 32] {
```

Comment:
created task as well:  https://axelarnetwork.atlassian.net/browse/AXE-2649

Reply:
Why not just `#[derive(Hash)]` on `Message`?

Reply:
for now I think it's ok to have a stable hashing function as a placeholder. We can address a finalized design in a followup PR

Reply:
was my first thought, but figured we would probably want to be in control of each individual field hash, specially if we change the message struct. Also I wasn't sure if it was a good idea to use the `Hash` trait since it's used by other things in Rust like HashMaps

Reply:
I am not sure that deriving Hash is safe for our use case: https://doc.rust-lang.org/std/hash/trait.Hash.html#portability

"Due to differences in endianness and type sizes, data fed by Hash to a Hasher should not be considered portable across platforms. Additionally the data passed by most standard library types should not be considered stable between compiler versions.

This means tests shouldn’t probe hard-coded hash values or data fed to a Hasher and instead should check consistency with Eq.

Serialization formats intended to be portable between platforms or compiler versions should either avoid encoding hashes or only rely on Hash and Hasher implementations that provide additional guarantees."

I worry that if we recompile the contract during an upgrade, the hashing might change. I do think it's very important to be absolutely sure the hashing function is stable.